biorxiv/pca_association_experiment/biorxiv_pca_plots.py

```diff
@@ -0,0 +1,639 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import plotnine as p9
+from PIL import ImageColor
+from scipy.spatial.distance import cdist
+from sklearn.decomposition import PCA
+from tqdm import tqdm_notebook
+import wordcloud
+
+
+# In[2]:
+
+
+def display_clouds(pc_cloud_1, pc_cloud_2):
+    return display(
+        HTML(
+            f"""
+            <table>
+                <tr>
+                    <td>
+                    <img src={pc_cloud_1}>
+                    </td>
+                    <td>
+                    <img src={pc_cloud_2}>
+                    </td>
+                </tr>
+            </table>
+            """
+        )
+    )
+
+
+# In[3]:
+
+
+journal_map_df = pd.read_csv("../exploratory_data_analysis/output/biorxiv_article_metadata.tsv", sep="\t")
+journal_map_df.head()
+
+
+# # PCA the Documents
+
+# Run PCA over the documents. Generates 50 principal components, but can generate more or less.
+
+# In[4]:
+
+
+n_components = 50
+random_state = 100
+
+
+# In[5]:
+
+
+biorxiv_articles_df = pd.read_csv(
+    "../word_vector_experiment/output/word2vec_output/biorxiv_all_articles_300.tsv.xz", 
+    sep="\t"
+)
+
+
+# In[6]:
+
+
+reducer = PCA(
+    n_components = n_components,
+    random_state = random_state
+)
+
+embedding = reducer.fit_transform(
+    biorxiv_articles_df[[f"feat_{idx}" for idx in range(300)]].values
+)
+
+pca_df = (
+    pd.DataFrame(embedding, columns=[f"pca{dim}" for dim in range(1, n_components+1, 1)])
+    .assign(document=biorxiv_articles_df.document.values.tolist())
+    .merge(journal_map_df[["category", "document", "doi"]], on="document")
+)
+
+latest_version = (
+    pca_df
+    .groupby("doi", as_index=False)
+    .agg({"document":"last"})
+    .document
+    .values
+    .tolist()
+)
+
+pca_df = (
+    pca_df
+    .query(f"document in {latest_version}")
+    .reset_index(drop=True)
+)
+
+
+# In[7]:
+
+
+reducer.explained_variance_
+
+
+# In[8]:
+
+
+reducer.explained_variance_ratio_
+
+
+# # Plot the PCA axes
+
+# This section visualizes PCA axes and attempts to provide an explanation for each plotted PC.
+# Give there are 50 pcs generated this notebook/section will only focus on PCs that generate a clear enough signal.
+
+# ## Basis PCs
+
+# When analyzing PCs, it is generally a good idea to start off with the PCs that are easily distinguishable and then branch out to PCs that are harder to interpret. This section focuses on PCs: 1,2,3,4 and 5.
+# 
+# These PCs correspond to the following concepts:
+# 
+# | PC | Concept (negative vs positive)|
+# | --- | --- |
+# | PCA1 | Quantitative Biology vs Molecular Biology |
+# | PCA2 | Neuroscience vs Genomics |
+# | PCA3 | Sequencing vs Disease |
+# | PCA4 | Microbiology vs Cell Biology |
+# | PCA5 | RNA-seq vs Evolutional Biology | 
+
+# ### PCA1 vs PCA2
+
+# In[9]:
+
+
+display_clouds(
+    'output/word_pca_similarity/figures/pca_01_cossim_word_cloud.png',
+    'output/word_pca_similarity/figures/pca_02_cossim_word_cloud.png',
+)
+
```

Comment:
Good question. Cosine similarity measures the angle between two vectors rather than distance, so it doesn't pay attention to outliers. I'd have to change metrics if I wanted to account for that.